extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -115,6 +115,12 @@ export let AdResponseDef;
     }} */
 let CreativeMetaDataDef;
 
+/** @typedef {{serviceName: string, keys: !Array<!Promise<?PublicKeyInfoDef>>}} */
+let SingleServiceKeySetInfoDef;
+
+/** @typedef {Array<!Promise<!SingleServiceKeySetInfoDef>>} */
+let PerServiceKeySetsInfoDef;
```

Comment:
Yeah, not my best naming job ever, I admit.  :-(  Tried to revise and also add comments to clarify.  Let me know if you still have concerns.